### ISSUE TITLE
VB -> C#:  Optional parameters must appear after all required parameters

### ISSUE THREAD

**Issue by BlackEagleXV** [icsharpcode/CodeConverter#908]
### VB.Net input code
```VB.NET
Friend Sub SpecialPrintList(section As Integer, Optional ByVal currentList As Integer = 0, Optional ByRef commCall As Boolean = False, Optional ByVal language As Integer = -1, Optional ByVal kindOfCall As Integer = 0)
```

### Erroneous output
```C#
internal void SpecialPrintList(int section, int currentList = 0, [Optional, DefaultParameterValue(false)] ref bool commCall, int language = -1, int kindOfCall = 0)
```

### Expected output
```C#
internal void SpecialPrintList(int section, [Optional, DefaultParameterValue(0)] int currentList, [Optional, DefaultParameterValue(false)] ref bool commCall, int language = -1, int kindOfCall = 0)
```

### Details
* Product in use: [https://github.com/icsharpcode/CodeConverter/pull/907](https://github.com/icsharpcode/CodeConverter/pull/907) VS extension
* Version in use: 73d55b29c6c1e09dd9e3b6e9b30073d639d9e00d
* Did you see it working in a previous version, which? No
* I don't know how to fix the default value problem of the ref argument. It can be that it is not fixable inside the method but on the caller side to always provide it. My issue is about the compiler error of the method. So don't focus on the ref argument if you don't want to.

**Comment by Yozer** [comment]
I wonder if I broke that in https://github.com/icsharpcode/CodeConverter/commit/73d55b29c6c1e09dd9e3b6e9b30073d639d9e00d or it never worked.
This case should be solved by hoisting the required argument in local variable.

**Comment by BlackEagleXV** [comment]
I guess you did not break it because the "official" version produces the same error. I just wanted to make sure to test the newest version before opening the bug...

**Comment by Yozer** [comment]
Yep, the same output on a version before my last change related to omitted args (tested on https://github.com/icsharpcode/CodeConverter/commit/b45ae61ef6676d6d450397b4607a81014d8abef7)
I will take a look at this since I still remember how the conversion for omitted args works.
Thanks for reporting!

**Comment by GrahamTheCoder** [comment]
Hmm I'd suggest it's best to keep them separate since the other is mainly about setting the default value (but mentions this issue), and this one is about ordering. Though if it all makes sense as one issue/fix to you then feel free.
